util.go

```diff
@@ -23,6 +23,23 @@ func isFalse(value interface{}) bool {
 	case nil:
 		return true
 	}
+	// Try the reflection cases before returning false.
+	vType := reflect.TypeOf(value).Kind()
+	if vType == reflect.Struct {
+		// A struct type will never be false, even if
+		// all of its values are the zero type.
+		return false
+	} else if vType == reflect.Slice {
+		if reflect.ValueOf(value).Len() == 0 {
+			return true
+		}
+		return false
+	} else if vType == reflect.Ptr {
```

Comment:
After checking if ptr this probably should also check for if the reflected value is `IsValid()` and `IsNil()`. It is possible to construct an interface value where it will not equal nil when not set. If the interface underlying is nil, but the interface is assigned an underlying type the interface value will not be nil, and in this case will panic.

```go
var a *int = nil
var myinf interface{}
myinf = a

// myinf != nil
```

http://play.golang.org/p/HLLdp4zDrW
https://golang.org/doc/faq#nil_error